src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayTooSmall(size);
```

Comment:
You can use size directly in method, thats delegation is redundant
```suggestion
        growIfArrayTooSmall();
```